yarn/src/main/scala/org/apache/spark/deploy/yarn/ExecutorRunnable.scala

```diff
@@ -290,10 +290,19 @@ class ExecutorRunnable(
       YarnSparkHadoopUtil.setEnvFromInputString(env, userEnvs)
     }
 
+    // lookup appropriate http scheme for container log urls
+    val yarnHttpPolicy = yarnConf.get(
+      YarnConfiguration.YARN_HTTP_POLICY_KEY,
```

Comment:
YARN_HTTP_POLICY_KEY and YARN_HTTP_POLICY_DEFAULT were part of the initial 2.2 release: [YARN-1277](https://issues.apache.org/jira/browse/YARN-1277)([commit](https://github.com/apache/hadoop/commit/21181b65531449e5fda321c11f0672c3067641aa#diff-6b291e39bccffdc169e769d0b09310c8R49)).